snmp/datadog_checks/snmp/utils.py

```diff
@@ -92,3 +95,69 @@ def oid_pattern_specificity(pattern):
         len(parts),  # Shorter OIDs are less specific than longer OIDs, regardless of their contents.
         parts,  # For same-length OIDs, compare their contents (integer parts).
     )
+
+
+class OIDPrinter(object):
```

Comment:
For each of the methods, it might be helpful to have a short example of what expected output might look like.  For instance, why would I want to choose `oid_str` vs `oid_str_value`?